### PR TITLE
Fixes #268.  Auto-detection of tiller will not support multiple tillers.

### LINKED ISSUE #268: Clusters with more than 1 tiller can produce unpredictable deploys
rok8s-scripts detects the tiller namespace and sets the `TILLER_NAMESPACE` variable accordingly.  However, when there is more than 1 tiller in a cluster it will pick the first one it finds.  When more than 1 tiller is present rok8s-scripts should produce an error instead of deploying.

### PR DESCRIPTION
**Pull request by sudermanjr**

### CODE REVIEW

File: bin/helm-deploy [hunk 1]
```diff
@@ -41,6 +37,21 @@ if [ "${HELM_TEMPLATE_ONLY}" != "true" ]; then
   helm init --client-only
 fi
 
+detect_tiller_namespace() {
+    # If there is more than one tiller, then auto-detection will not work.
+    mapfile -t TILLERS < <(kubectl get deploy --all-namespaces -l app=helm,name=tiller | awk 'NR > 1 {print $1}')
+    if [ ${#TILLERS[@]} -ne 1 ]; then
```
Comment (bobby-brennan) [inline bin/helm-deploy]: Should probably be `-gt 1`, since the zero case is handled below

Comment (sudermanjr) [reply bin/helm-deploy]: 👍 Fixed

File: bin/helm-deploy [hunk 2]
```diff
@@ -41,6 +37,21 @@ if [ "${HELM_TEMPLATE_ONLY}" != "true" ]; then
...
+    # If there is more than one tiller, then auto-detection will not work.
+    mapfile -t TILLERS < <(kubectl get deploy --all-namespaces -l app=helm,name=tiller | awk 'NR > 1 {print $1}')
+    if [ ${#TILLERS[@]} -gt 1 ]; then
+      echo "Multiple TILLERS were found.  This is not supported by auto-detection.  Please specify TILLER_NAMESPACE in your config."
+      exit 1
+    else
+      TILLER_NAMESPACE="${TILLERS[0]}"
+
+      # just in case variable didn't get set, give it a default
+      TILLER_NAMESPACE=${TILLER_NAMESPACE:-kube-system}
```
Comment (lanceops) [inline bin/helm-deploy]: Not a show stopper but is it possible to set the default in line 47?

Comment (sudermanjr) [reply bin/helm-deploy]: Totally.  Good call.

File: bin/helm-deploy [hunk 3]
```diff
@@ -41,6 +37,18 @@ if [ "${HELM_TEMPLATE_ONLY}" != "true" ]; then
   helm init --client-only
 fi
 
+detect_tiller_namespace() {
+    # If there is more than one tiller, then auto-detection will not work.
+    mapfile -t TILLERS < <(kubectl get deploy --all-namespaces -l app=helm,name=tiller | awk 'NR > 1 {print $1}')
+    if [ ${#TILLERS[@]} -gt 1 ]; then
+      echo "Multiple TILLERS were found.  This is not supported by auto-detection.  Please specify TILLER_NAMESPACE in your config."
```
Comment (endzyme) [inline bin/helm-deploy]: Only suggestion (non-blocking) is to list the namespaces in which tiller was found - so the user can adjust without as much investigation.

**Comment by endzyme** [comment]
I would also assume this could be a breaking change to some installations, which might be a consideration for the version bump.

File: bin/helm-deploy (hunk 3 above)
Comment (sudermanjr) [reply bin/helm-deploy]: 👍 Done.

**Comment by sudermanjr** [comment]
@endzyme The version question is an interesting one.  I feel like anyone who was using this in a multi-tiller environment without setting the TILLER_NAMESPACE variable directly was getting extremely unreliable behavior.

However, this could cause pipelines that work currently to stop working, so perhaps a minor version bump is in order.

**Pull request merged**